src/plugin_system/plugin_system.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * aaltitoad - a verification engine for tick tock automata models
+   Copyright (C) 2023 Asger Gitz-Johansen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include <util/warnings.h>
+#include "plugin_system.h"
+
+namespace aaltitoad::plugins {
+    template<typename T>
+    T load_symbol(void* handle, const std::string& symbol_name) {
+        T val = (T) dlsym(handle, symbol_name.c_str());
+        if(!val)
+            throw std::logic_error("could not find "+symbol_name+" symbol");
+        return val;
+    }
+
+    bool is_dynamic_library(const std::string& filename) {
```

Comment:
This function is a bit naive. It works for now, but it would be a great "help wanted" issue

Reply:
Created #40